ci/infra/testrunner/tests/test_node_reboot.py

```diff
@@ -0,0 +1,71 @@
+import logging
+import json
+import time
+
+import pytest
+from timeout_decorator import timeout
+
+logger = logging.getLogger("testrunner")
+
+
+@timeout(120)
+def wait_for_node_reboot(node_type, node_number, platform):
+    while True:
+        try:
+            platform.ssh_run(node_type, node_number, 'echo hello')
+        except Exception as ex:
+            logger.debug(ex)
+        else:
+            break
+
+        time.sleep(5)
+
+
+@timeout(120)
+def check_nodes_ready(kubectl):
+    while True:
+        try:
+            kubectl.run_kubectl('wait --for=condition=Ready node --all')
+        except Exception as ex:
+            logger.debug(ex)
+        else:
+            break
+
+        time.sleep(5)
+
+
+@timeout(120)
+def check_pods_ready(kubectl):
+    while True:
+        try:
+            pods = json.loads(kubectl.run_kubectl('get pods --all-namespaces -o json'))['items']
+
+            for pod in pods:
+                pod_status = pod['status']['phase']
+                assert pod_status == 'Running', f'Pod {pod["metadata"]["name"]} status {pod_status} != Running'
+        except Exception as ex:
+            logger.debug(ex)
+        else:
+            break
+
+        time.sleep(5)
+
+
```

Comment:
This is a great suggestion. This has been asked by some developers for other tests. I think we could add this to the development library so that it can be used by all tests.

Reply:
yes this would be very helpful for some scenarios in the e2e tests, it would be great if we could have a generic repeater function

Reply:
It should only require to move these two functions to either ' testconf` file or create another ' testlib.py'  file.